components/forms/PostForm.js

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import Form from 'react-bootstrap/Form';
+import FloatingLabel from 'react-bootstrap/FloatingLabel';
+import PropTypes from 'prop-types';
+import { Button } from 'react-bootstrap';
+import { getAllCategories } from '../../api/tagData';
+import { createPost, updatePost } from '../../api/postsData';
+
+const initialState = {
+  id: 0,
+  userId: 0,
+  categoryId: 0,
+  title: '',
+  publicationDate: '',
+  imageUrl: '',
+  content: '',
+  approved: false,
+};
+
+function PostForm({ obj }) {
+  const [formInput, setFormInput] = useState(initialState);
+  // const [tags, setTags] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const router = useRouter();
+
+  useEffect(() => {
+  // getAllTags().then(setTags);
+    getAllCategories().then(setCategories);
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormInput((prevState) => ({
+      ...prevState,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = (p) => {
+    p.preventDefault();
+    // ADDED PAYLOAD VARIABLE, ADDED USERID AND PUBLICATION DATE KEYS - NEED TO FETCH USER ID AND REPLACE THE TEST CONSTANT BELOW; NEED TO ALSO ENSURE PUBLICATIONDATE IS PULLING IN CORRECT DATE/TIME DATA, CURRENT VALUE IS A DUMMY FOR TESTING
+    const payload = { ...formInput, userId: 1, publicationDate: Date.now };
+    console.warn(payload);
+    if (obj.id) {
+      updatePost(payload).then(() => router.push('/posts'));
+    } else {
+      createPost(payload).then(() => router.push('/posts'));
+    }
+  };
+
+  return (
+    <Form onSubmit={handleSubmit}>
+      <h2 className="text-white mt-5">{obj.id ? 'Update' : 'Create'} Post</h2>
+
```

Comment:
We can insert a user dropdown to assign the right userId/user to the post.      
<FloatingLabel controlId="floatingSelect" label="User">
        <Form.Select
          aria-label="User"
          name="userId"
          onChange={handleChange}
          className="mb-3"
          // CHANGED KEY TO FORMINPUT VERSUS OBJ
          value={formInput.userId}
          required
        >
          <option value="">Select the user creating the Post</option>
          {
          users.map((u) => (
            <option
              key={u.id}
              value={u.id}
            >
              {u.firstName} {u.lastName}
            </option>
          ))
        }
        </Form.Select>
      </FloatingLabel>